### PR TITLE
feat: add search link to help modal

### LINKED ISSUE #47375: Suggest searching the forum before making a help thread
### Describe the Issue

We should find a way to cut down on the number of support threads. Not just by improving the challenges but by improving the help the camper can find for themselves before even having to ask a question.

I think if we ask the camper to search the forum when making an "ask for help" thread it might do the trick. Or now that the forum can handle the same title for multiple threads maybe we can automatically link to a search using the challenge title?

Either one I think can help cut down the number of support threads.

### Affected Page

https://forum.freecodecamp.org/

### Your code

```

```

### Expected behavior

Less of the same threads.

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #47375 

Thought I'd have a play with this. This adds a link to the search page of the forum with a pre-filled query based on the current challenge. The idea was to encourage searching for a solution to their question without making the experience of asking for help frustrating.

### CODE REVIEW

File: client/src/templates/Challenges/components/help-modal.tsx [hunk 1]
```diff
@@ -31,12 +33,23 @@ const mapDispatchToProps = (dispatch: Dispatch) =>
 
 const RSA = forumLocation + '/t/19514';
 
+const generateSearchLink = (title: string, block: string) => {
+  console.table({ title, block });
```
Comment (lasjorg) [inline client/src/templates/Challenges/components/help-modal.tsx:37]: Is the console log left in intentional?

**Comment by lasjorg** [comment]
I tested it briefly and it works as intended, which is great.

I would however suggest making it more attention-grabbing both visually and verbally to make sure it isn't ignored. Maybe some colors and some bold text might help as well or an icon (like an exclamation icon). Maybe even a hover effect where hovering the create thread button will highlight the link or something like that.

Anyway, I like it, I just wish it wasn't so easy to ignore.

File: client/src/templates/Challenges/components/help-modal.tsx (hunk 1 above)
Comment (naomi-lgbt) [reply client/src/templates/Challenges/components/help-modal.tsx:37]: Nope - I'm very good at forgetting to remove these 😅

**Comment by naomi-lgbt** [comment]
image

I've added an exclamation and some bolded text.

**Comment by lasjorg** [comment]
Definitely better. It might need to be stacked so the icon is centered above the text for small screens.

I might also suggest moving the end of the sentence to the start.

> **Before making a post**, please...

Not sure about the verbiage or how to make the link more obvious but I was thinking something in the ballpark of this.

Desktop
modal-desktop

Mobile
modal-mobile

**Comment by naomi-lgbt** [comment]
I completely forgot about bootstrap's alert classes.

image

image

image

**Comment by lasjorg** [comment]
Although I don't really mind it being a little aggressive (lol) but maybe we should use the info class instead of danger? I'm fine with either.

Not to be a pain, but can you add 10px margin-bottom to the icon to get the spacing more even?

**Comment by naomi-lgbt** [comment]
I don't want to be *too* aggressive, but at the same time I want campers to actually read it lol.

> Not to be a pain, but can you add 10px margin-bottom to the icon to get the spacing more even?

Not a pain at all - I appreciate all of your feedback it's been a huge help.

**Comment by naomi-lgbt** [comment]
Info:

image

Warning:

image

**Comment by lasjorg** [comment]
Hmm, not sure.

Maybe we just keep it the danger class, at least that way people have no excuse for not reading it. The info color is less in-you-face but as you said we do want people to actually read the message.

**Comment by naomi-lgbt** [comment]
I might ping @ahmadabdolsaheb on this - it could be worth doing some A/B testing.

Gonna push up the padding fix but leave the class alone for now.

**Comment by scissorsneedfoodtoo** [comment]
Just tested locally and this looks great. Thanks for tackling this @naomi-lgbt, and for all of your helpful suggestions @lasjorg.

Tried out the modal with a few of the different types of challenges, and tested in a couple of other languages. If there are posts on the forum with the same or similar text as the search query from the help modal, they surface on the forum's search results page. And it's good that learners have the ability to add filters and narrow their search if there are a lot of hits.

Everything LGTM 👍

Also, I can see the benefit of using the danger styling to grab people's attention to the new string and link. But I can also see reserving that styling for errors.

Interested to hear what @ahmadabdolsaheb and others think about that.

**Review by scissorsneedfoodtoo — APPROVED** [review]

**Review by raisedadead — APPROVED** [review]

**Review by lasjorg — APPROVED** [review]
LGTM.

Let's keep the danger class for now and see how it works. If we think it needs to be changed later it's a simple class switch.

**Comment by ahmadabdolsaheb** [comment]
Sorry for being late to the review, solid work. Attention or Error would both work depending on the urgency.
The only nitpick is that the link's hover color does not match the background color. Swapping the background and font color should do it.

**Pull request merged**